sql/thriftserver/src/main/scala/org/apache/spark/sql/thriftserver/cli/RowBasedSet.scala

```diff
@@ -0,0 +1,154 @@
+        throw new IllegalArgumentException("The given schema does't match the given row")
+      }
+      rows.map(toTRow)
+    } else {
+      Nil
+    }
+  }
+
+  private[this] def toTRow(row: Row): TRow = {
+    val tRow = new TRow()
+    (0 until row.length).map(i => toTColumnValue(i, row)).foreach(tRow.addToColVals)
+    tRow
+  }
+
+  private[this] def toTColumnValue(ordinal: Int, row: Row): TColumnValue =
+    types(ordinal).dataType match {
+      case BooleanType =>
+        val boolValue = new TBoolValue
+        if (!row.isNullAt(ordinal)) boolValue.setValue(row.getBoolean(ordinal))
+        TColumnValue.boolVal(boolValue)
+
+      case ByteType =>
+        val byteValue = new TByteValue
+        if (!row.isNullAt(ordinal)) byteValue.setValue(row.getByte(ordinal))
+        TColumnValue.byteVal(byteValue)
+
+      case ShortType =>
+        val tI16Value = new TI16Value
+        if (!row.isNullAt(ordinal)) tI16Value.setValue(row.getShort(ordinal))
+        TColumnValue.i16Val(tI16Value)
+
+      case IntegerType =>
+        val tI32Value = new TI32Value
+        if (!row.isNullAt(ordinal)) tI32Value.setValue(row.getInt(ordinal))
+        TColumnValue.i32Val(tI32Value)
+
+      case LongType =>
+        val tI64Value = new TI64Value
+        if (!row.isNullAt(ordinal)) tI64Value.setValue(row.getLong(ordinal))
+        TColumnValue.i64Val(tI64Value)
+
+      case FloatType =>
+        val tDoubleValue = new TDoubleValue
+        if (!row.isNullAt(ordinal)) tDoubleValue.setValue(row.getFloat(ordinal))
+        TColumnValue.doubleVal(tDoubleValue)
+
+      case DoubleType =>
+        val tDoubleValue = new TDoubleValue
+        if (!row.isNullAt(ordinal)) tDoubleValue.setValue(row.getDouble(ordinal))
+        TColumnValue.doubleVal(tDoubleValue)
+
+      case StringType =>
+        val tStringValue = new TStringValue
+        if (!row.isNullAt(ordinal)) tStringValue.setValue(row.getString(ordinal))
+        TColumnValue.stringVal(tStringValue)
+
+      case DecimalType() =>
+        val tStrValue = new TStringValue
+        if (!row.isNullAt(ordinal)) tStrValue.setValue(row.getDecimal(ordinal).toString)
+        TColumnValue.stringVal(tStrValue)
+
+      case DateType =>
+        val tStringValue = new TStringValue
+        if (!row.isNullAt(ordinal)) tStringValue.setValue(row.get(ordinal).toString)
+        TColumnValue.stringVal(tStringValue)
+
+      case TimestampType =>
+        val tStringValue = new TStringValue
+        if (!row.isNullAt(ordinal)) tStringValue.setValue(row.get(ordinal).toString)
+        TColumnValue.stringVal(tStringValue)
+
+      case BinaryType =>
+        val tStringValue = new TStringValue
+        if (!row.isNullAt(ordinal)) {
+          val bytes = row.getAs[Array[Byte]](ordinal)
+          tStringValue.setValue(HiveResult.toHiveString((bytes, types(ordinal).dataType)))
+        }
+        TColumnValue.stringVal(tStringValue)
+
+      case _: ArrayType | _: StructType | _: MapType =>
```

Comment:
Compared to https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala#L83
I think this is missing `_: UserDefinedType[_]` here, and also a match for `CalendarIntervalType`, and I think also `NullType`.
Same in ColumnBasedSet.

Reply:
> Compared to https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala#L83
> I think this is missing `_: UserDefinedType[_]` here, and also a match for `CalendarIntervalType`, and I think also `NullType`.
> Same in ColumnBasedSet.

Yeah, update RowBasedSet and ColumnBasedSet